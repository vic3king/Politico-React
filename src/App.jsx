import React, { Component } from 'react';
import { Provider } from 'react-redux';
import Notifications from 'react-notify-toast';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import store from './store/store';
import ErrorPage from './components/NotFound-404/Pagenotfound';
import HomePage from './components/Home/Home';
import Signup from './components/Registration/Signup';
import Login from './container/login.container';
import Logout from './components/Logout/Logout';
import CitizensPage from './container/citizens-page.container';
import AdminPage from './container/admin-page.container';
import PoliticiansPage from './container/politicians-page.container';

class App extends Component {
  state = {};

  render() {
    return (
      <Provider store={store}>
        <React.Fragment>
          <Notifications />
          <BrowserRouter>
            <Switch>
              <Route path="/" exact component={HomePage} />
              <Route path="/signup" component={Signup} />
              <Route path="/login" component={Login} />
              <Route path="/admin-dashboard" component={AdminPage} />
              <Route
                path="/politicians-dashboard"
                component={PoliticiansPage}
              />
              <Route path="/citizens-dashboard" component={CitizensPage} />
              <Route path="/logout" component={Logout} />
              <Route path="/error" component={ErrorPage} />
              <Route path="*" component={ErrorPage} />
              <Redirect to="/error" />
            </Switch>
          </BrowserRouter>
        </React.Fragment>
      </Provider>
    );
  }
}

export default App;
