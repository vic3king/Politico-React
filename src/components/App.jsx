import React, { Component } from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import ErrorPage from './NotFound-404/Pagenotfound';
import HomePage from './Home/Home';
import Signup from './Registration/Signup';
import Login from './Registration/Login';
import AdminPage from './Dashboard/Admin/Admin';
import ResultsPage from './Results/ResultsPage';

class App extends Component {
  state = {};

  render() {
    return (
      <React.Fragment>
        <BrowserRouter>
          <Switch>
            <Route path="/" exact component={HomePage} />
            <Route path="/signup" component={Signup} />
            <Route path="/login" component={Login} />
            <Route path="/admin-dashboard" component={AdminPage} />
            <Route path="/results" component={ResultsPage} />
            <Route path="/error" component={ErrorPage} />
            <Route path="*" component={ErrorPage} />
            <Redirect to="/error" />
          </Switch>
        </BrowserRouter>
      </React.Fragment>
    );
  }
}

export default App;
