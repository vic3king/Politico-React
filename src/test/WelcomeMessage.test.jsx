import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';
import HomePage from '../components/Home/Home';

describe('Home component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(<HomePage />));
  });

  it('renders correctly', () => {
    const wrapper = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    ).toJSON;
    expect(wrapper).toMatchInlineSnapshot(`[Function]`);
  });
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
