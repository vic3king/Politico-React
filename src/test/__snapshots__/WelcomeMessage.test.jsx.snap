// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <BrowserRouter>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/signup"
          />
          <Route
            component={[Function]}
            path="/login"
          />
          <Route
            component={[Function]}
            path="/admin-dashboard"
          />
          <Route
            component={[Function]}
            path="/politicians-dashboard"
          />
          <Route
            component={[Function]}
            path="/citizens-dashboard"
          />
          <Route
            component={[Function]}
            path="/results"
          />
          <Route
            component={[Function]}
            path="/logout"
          />
          <Route
            component={[Function]}
            path="/error"
          />
          <Route
            component={[Function]}
            path="*"
          />
          <Redirect
            to="/error"
          />
        </Switch>
      </BrowserRouter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/signup"
          />
          <Route
            component={[Function]}
            path="/login"
          />
          <Route
            component={[Function]}
            path="/admin-dashboard"
          />
          <Route
            component={[Function]}
            path="/politicians-dashboard"
          />
          <Route
            component={[Function]}
            path="/citizens-dashboard"
          />
          <Route
            component={[Function]}
            path="/results"
          />
          <Route
            component={[Function]}
            path="/logout"
          />
          <Route
            component={[Function]}
            path="/error"
          />
          <Route
            component={[Function]}
            path="*"
          />
          <Redirect
            to="/error"
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/signup"
            />,
            <Route
              component={[Function]}
              path="/login"
            />,
            <Route
              component={[Function]}
              path="/admin-dashboard"
            />,
            <Route
              component={[Function]}
              path="/politicians-dashboard"
            />,
            <Route
              component={[Function]}
              path="/citizens-dashboard"
            />,
            <Route
              component={[Function]}
              path="/results"
            />,
            <Route
              component={[Function]}
              path="/logout"
            />,
            <Route
              component={[Function]}
              path="/error"
            />,
            <Route
              component={[Function]}
              path="*"
            />,
            <Redirect
              to="/error"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/admin-dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/politicians-dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/citizens-dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/results",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/logout",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/error",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "*",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "to": "/error",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/signup"
            />
            <Route
              component={[Function]}
              path="/login"
            />
            <Route
              component={[Function]}
              path="/admin-dashboard"
            />
            <Route
              component={[Function]}
              path="/politicians-dashboard"
            />
            <Route
              component={[Function]}
              path="/citizens-dashboard"
            />
            <Route
              component={[Function]}
              path="/results"
            />
            <Route
              component={[Function]}
              path="/logout"
            />
            <Route
              component={[Function]}
              path="/error"
            />
            <Route
              component={[Function]}
              path="*"
            />
            <Redirect
              to="/error"
            />
          </Switch>
        </BrowserRouter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/signup"
            />
            <Route
              component={[Function]}
              path="/login"
            />
            <Route
              component={[Function]}
              path="/admin-dashboard"
            />
            <Route
              component={[Function]}
              path="/politicians-dashboard"
            />
            <Route
              component={[Function]}
              path="/citizens-dashboard"
            />
            <Route
              component={[Function]}
              path="/results"
            />
            <Route
              component={[Function]}
              path="/logout"
            />
            <Route
              component={[Function]}
              path="/error"
            />
            <Route
              component={[Function]}
              path="*"
            />
            <Redirect
              to="/error"
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/signup"
              />,
              <Route
                component={[Function]}
                path="/login"
              />,
              <Route
                component={[Function]}
                path="/admin-dashboard"
              />,
              <Route
                component={[Function]}
                path="/politicians-dashboard"
              />,
              <Route
                component={[Function]}
                path="/citizens-dashboard"
              />,
              <Route
                component={[Function]}
                path="/results"
              />,
              <Route
                component={[Function]}
                path="/logout"
              />,
              <Route
                component={[Function]}
                path="/error"
              />,
              <Route
                component={[Function]}
                path="*"
              />,
              <Redirect
                to="/error"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/admin-dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/politicians-dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/citizens-dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/results",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/logout",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/error",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "*",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "to": "/error",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
